File: backend-v5/lms/poll/PollResult.php
<?php

// Modul: lms/poll
// Klasa: PollResult
// Opis: jedan popunjen anketni listić

// TODO naći bolji naziv od result "listić?"

require_once(Config::$backend_path."core/DB.php");

class PollResult {
	public $id;
	public $pollId, $time, $hashCode, $finished, $courseUnitId, $academicYearId, $programmeId, $semester, $studentId; // FIXME većina ovih podataka su nepotrebni, treba odrediti tačno koji

	public static function fromId($id) {
		$q10 = DB::query("select anketa, vrijeme, unique_id, zavrsena, predmet, akademska_godina, studij, semestar, student from anketa_rezultat where id=$id");
		if (mysql_num_rows($q10)<1) {
			throw new Exception("unknown poll result");
		}
		
		$p = new PollResult;
		$p->id = $id;
		$p->pollId = mysql_result($q10,0,0);
		$p->time = mysql_result($q10,0,1);
		$p->hashCode = mysql_result($q10,0,2);
		if (mysql_result($q10,0,3) == "Y") $p->finished=true; else $p->finished=false;
		$p->courseUnitId = mysql_result($q10,0,4);
		$p->academicYearId = mysql_result($q10,0,5);
		$p->programmeId = mysql_result($q10,0,6);
		$p->semester = mysql_result($q10,0,7);
		$p->studentId = mysql_result($q10,0,8);
		
		return $p;
	}

	public static function fromHash($hashCode) {
		$q10 = DB::query("select anketa, vrijeme, id, zavrsena, predmet, akademska_godina, studij, semestar, student from anketa_rezultat where id=$id");
		if (mysql_num_rows($q10)<1) {
			throw new Exception("unknown poll result");
		}
		
		$p = new PollResult;
		$p->hashCode= $hashCode;
		$p->pollId = mysql_result($q10,0,0);
		$p->time = mysql_result($q10,0,1);
		$p->id = mysql_result($q10,0,2);
		if (mysql_result($q10,0,3) == "Y") $p->finished=true; else $p->finished=false;
		$p->courseUnitId = mysql_result($q10,0,4);
		$p->academicYearId = mysql_result($q10,0,5);
		$p->programmeId = mysql_result($q10,0,6);
		$p->semester = mysql_result($q10,0,7);
		$p->studentId = mysql_result($q10,0,8);
		
		return $p;
	}
	
	public static function fromStudentAndPoll($studentId, $pollId) {
		$q10 = DB::query("select id, vrijeme, unique_id, zavrsena, predmet, akademska_godina, studij, semestar from anketa_rezultat where anketa=$pollId and student=$studentId");
		if (mysql_num_rows($q10)<1) {
			throw new Exception("unknown poll result");
		}
		
		$p = new PollResult;
		$p->id = mysql_result($q10,0,0);
		$p->pollId = $pollId;
		$p->time = mysql_result($q10,0,1);
		$p->hashCode = mysql_result($q10,0,2);
		if (mysql_result($q10,0,3) == "Y") $p->finished=true; else $p->finished=false;
		$p->courseUnitId = mysql_result($q10,0,4);
		$p->academicYearId = mysql_result($q10,0,5);
		$p->programmeId = mysql_result($q10,0,6);
		$p->semester = mysql_result($q10,0,7);
		$p->studentId = $studentId;
		
		return $p;		
	}
	
	// Add new PollResult item to database
	public function add() {
		if ($this->finished) $finsql="'Y'"; else $finsql="'N'";
		$q10 = DB::query("insert into anketa_rezultat set anketa=".$this->pollId.", vrijeme=NOW(), unique_id='".$this->hashCode."', zavrsena=$finsql, predmet=".$this->courseUnitId.", akademska_godina=".$this->academicYearId.", studij=".$this->programmeId.", semestar=".$this->semester.", student=".$this->studentId);
		$this->id = mysql_insert_id();
		$q20 = DB::query("select UNIX_TIMESTAMP(vrijeme) from anketa_rezultat where id=".$this->id);
		$this->time = mysql_result($q20,0,0);
	}
	
	// Update PollResult
	public function update() {
		if ($this->id < 1) throw new Exception("id not set, can't update");
		
		if ($this->finished) $finsql="'Y'"; else $finsql="'N'";
		$q10 = DB::query("update anketa_rezultat set anketa=".$this->pollId.", vrijeme=FROM_UNIXTIME(".$this->time."), unique_id='".$this->hashCode."', zavrsena=$finsql, predmet=".$this->courseUnitId.", akademska_godina=".$this->academicYearId.", studij=".$this->programmeId.", semestar=".$this->semester.", student=".$this->studentId." where id=".$this->id);
	}
}

?>

File: rorbackend/app/models/lms/poll/poll_result.rb
class Lms::Poll::PollResult < ActiveRecord::Base
  # Uncomment following lines if working with legacy database
  # set_table_name 'anketa_rezultat'
  # set_primary_key :id
  # alias_attribute :id, :id
  # alias_attribute :poll_id, :anketa
  # alias_attribute :time, :vrijeme
  # alias_attribute :closed, :zavrsena
  # alias_attribute :course_unit_id, :predmet
  # alias_attribute :unique_id, :unique_id
  # alias_attribute :academic_year_id, :akademska_godina
  # alias_attribute :programme_id, :studij
  # alias_attribute :semester, :semestar

  # Uncomment following lines if working with legacy database
  # TABLE_NAME = 'anketa_rezultat'
  # ID = TABLE_NAME + '.' + 'id'
  # POLL_ID = TABLE_NAME + '.' + 'anketa'
  # TIME = TABLE_NAME + '.' + 'vrijeme'
  # CLOSED = TABLE_NAME + '.' + 'zavrsena'
  # COURSE_UNIT_ID = TABLE_NAME + '.' + 'predmet'
  # UNIQUE_ID = TABLE_NAME + '.' + 'unique_id'
  # ACADEMIC_YEAR_ID = TABLE_NAME + '.' + 'akademska_godina'
  # PROGRAMME_ID = TABLE_NAME + '.' + 'studij'
  # SEMESTER = TABLE_NAME + '.' + 'semestar'

  # Comment following lines if working with legacy database
  TABLE_NAME = 'lms_poll_poll_results'
  ID = TABLE_NAME + '.' + 'id'
  POLL_ID = TABLE_NAME + '.' + 'poll_id'
  TIME = TABLE_NAME + '.' + 'time'
  CLOSED = TABLE_NAME + '.' + 'closed'
  COURSE_UNIT_ID = TABLE_NAME + '.' + 'course_unit_id'
  UNIQUE_ID = TABLE_NAME + '.' + 'unique_id'
  ACADEMIC_YEAR_ID = TABLE_NAME + '.' + 'academic_year_id'
  PROGRAMME_ID = TABLE_NAME + '.' + 'programme_id'
  SEMESTER = TABLE_NAME + '.' + 'semester'

  ALL_COLUMNS = [ID, POLL_ID, TIME, CLOSED, COURSE_UNIT_ID, UNIQUE_ID, ACADEMIC_YEAR_ID, PROGRAMME_ID, SEMESTER]
  
  
  belongs_to :course_unit, :class_name => "Core::CourseUnit"
  belongs_to :academic_year, :class_name => "Core::AcademicYear"
  belongs_to :programme, :class_name => "Core::Programme"
  
  
  def self.from_hash(unique_id)
    poll_result = (Lms::Poll::PollResult).where(:unique_id => unique_id)
    return poll_result
  end
  
  
  def self.from_student_and_poll(poll_id, student_id)
    poll_results = (Lms::Poll::PollResult).where(:poll_id => poll_id, :student_id => student_id)
    # TODO Gdje je kolona student??
    return poll_results
  end
  
  
  
end
